### PR TITLE
Add CI for Javadocs

### PR DESCRIPTION
**Pull request by CarlAmko**
Resolves https://github.com/ngrok-private/ngrok/issues/20185

**Validation**
https://ngrok.github.io/ngrok-java/

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,9 +21,9 @@ jobs:
       - uses: ./.github/workflows/rust-cache
       - uses: actions/setup-java@v3
         with:
-          java-version: '17'
-          distribution: 'temurin'
-          cache: 'maven'
+          java-version: "17"
```
Comment (nikolay-ngrok) [inline .github/workflows/ci.yml]: why did you change to `"`?

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+name: Javadocs
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```
Comment (nikolay-ngrok) [inline .github/workflows/docs.yml]: do we want this on pull requests?

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          java-version: "17"
+          distribution: "temurin"
+          cache: "maven"
+      - name: Build Javadocs
+        shell: sh
```
Comment (nikolay-ngrok) [inline .github/workflows/docs.yml]: why do we need `shell` here?

File: .github/workflows/docs.yml (hunk 2 above)
Comment (CarlAmko) [reply .github/workflows/docs.yml]: Hmm, I didn't think it was problematic. This way we could catch errors in the CI / javadocs before it hits `main`. 🤷🏼

Comment (nikolay-ngrok) [reply .github/workflows/docs.yml]: so, each PR could push a new version of the docs, even if not getting merged?

Comment (CarlAmko) [reply .github/workflows/docs.yml]: To prevent abuse, we can remove it.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (CarlAmko) [reply .github/workflows/ci.yml]: YAML linter.

File: .github/workflows/docs.yml (hunk 2 above)
Comment (CarlAmko) [reply .github/workflows/docs.yml]: Oh and also to test that this even worked. Without it on PRs, we wouldn't know that the pipeline works 😄 

The tradeoff here is that we can't correct any Javadocs errors until _after_ it merges into `main`.

Comment (CarlAmko) [reply .github/workflows/docs.yml]: One idea here might be to add in an [action like this](https://github.com/marketplace/actions/pull-request-comment-trigger). Then we can "manually" trigger the docs CI on a per-PR basis.

**Review by nikolay-ngrok — APPROVED** [review]
👍
